presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/ReduceAggStateFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+
+public class ReduceAggStateFactory
+        implements AccumulatorStateFactory<ReduceAggState>
+{
+    @Override
+    public ReduceAggState createSingleState()
+    {
+        return new SingleState();
+    }
+
+    @Override
+    public Class<? extends ReduceAggState> getSingleStateClass()
+    {
+        return SingleState.class;
+    }
+
+    @Override
+    public ReduceAggState createGroupedState()
+    {
+        return new GroupedState();
+    }
+
+    @Override
+    public Class<? extends ReduceAggState> getGroupedStateClass()
+    {
+        return GroupedState.class;
+    }
+
+    public static class GroupedState
+            extends AbstractGroupedAccumulatorState
+            implements ReduceAggState
+    {
+        private final ObjectBigArray<Object> values = new ObjectBigArray<>();
+
+        @Override
+        public long getEstimatedSize()
+        {
+            return values.sizeOf();
```

Comment:
This might be a bug. This is only tracking the ObjectBigArray<> size. We are not tracking the Object sizes. Can we add a unit test for memory accounting?

Reply:
Done.